napari/components/_viewer_key_bindings.py

```diff
@@ -59,15 +58,20 @@ def toggle_ndisplay(viewer: Viewer):
 @register_viewer_action(trans._("Toggle theme."))
 def toggle_theme(viewer: Viewer):
     """Toggle theme for viewer"""
-    settings = get_settings()
     themes = available_themes()
-    current_theme = settings.appearance.theme
+    current_theme = viewer.theme
+    # Check what the system theme is, to toggle properly
+    if current_theme == 'system':
+        current_theme = get_system_theme()
     idx = themes.index(current_theme)
     idx += 1
-    if idx == len(themes):
+    # Don't toggle to system, just among actual themes
+    if themes[idx] == 'system':
+        idx += 1
```

Comment:
I don't have a real opinion on this, just highlighting because it seems to be a change. Why this?

Reply:
It fixes an annoyance:
If you have the normal `light`, `dark`, `system`, lets say you're in Dark, the default, and you want to switch your theme to Light to test something or show something.
Live: you press the toggle keybind, but you don't get Light, you get `system`. Depending on the time of day, you may or may not get a change of theme. Lets say you do get a switch to `light`, because it's daytime. Ok, you're happy—all of this was behind the scenes.
Done with your screenshots or whatever, you want to go back to Dark, so you hit the toggle. What do you get? Not Dark, but `light`, because you weren't in Light, you were in `system`.
So this snippet just skips `system`, so you will now only toggle between actual themes.

Reply:
Ok. I see that I'm not writing everything. in line 67 there is the increment of `idx`. After this, you check `themes[idx]`. And in this place, you could get the index error. 
So before this check also should be verified. 

During the test you do not hit the problem because `"system"` theme is at the last position:

https://github.com/napari/napari/blob/8bb7e78f8fa9ec05fa82302c9e27254a85c1cc11/napari/utils/theme.py#L274

But if you change the theme to the `system` in the settings of napari and then call this action then you should expect an exception. 

But of course, the next if is required as there will be (for the current implementation of themes order) need to change it to zero.